Accompany/Controllers/Core/Home/AddNoteDetailsTableViewController.swift

```diff
@@ -0,0 +1,131 @@
+//
+//  SemesterDetailsTableViewController.swift
+//  Accompany
+//
+//  Created by Kelbin David on 2022-02-10.
+//
+
+import UIKit
+import SnapKit
+
+protocol AddNoteDetailsTableViewControllerDelegate {
+  func add(todo: ToDo)
+  func edit(todo: ToDo)
+}
+
+class AddNoteDetailsTableViewController: UITableViewController {
+  
+  var saveBarButton: UIBarButtonItem = {
+    let barButton = UIBarButtonItem(barButtonSystemItem: .save, target: self, action: #selector(saveButtonTapped))
+    return barButton
+  }()
+  
+  var todo: ToDo?
+  var delegate: AddNoteDetailsTableViewControllerDelegate?
+  var isCompleteButton: UIButton?
+  var toDoTitleTextField: UITextField?
+  var noteTextView: UITextView?
+  let buttonIndexPath = IndexPath(row: 0, section: 0)
+  let noteIndexPath = IndexPath(row: 0, section: 1)
+  
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    self.navigationItem.rightBarButtonItem = saveBarButton
+    if todo != nil {
+      navigationItem.title = "Edit To Do"
+    } else {
+      navigationItem.title = "Add To Do"
+    }
+    tableView.register(AddNoteCellOneTableViewCell.self, forCellReuseIdentifier: "cell")
+    tableView.register(AddNoteCellTwoTableViewCell.self, forCellReuseIdentifier: "cell2")
+    tableView.isUserInteractionEnabled = true
+    let notificationCenter = NotificationCenter.default
+    notificationCenter.addObserver(self, selector: #selector(textViewDidChange), name: UITextView.textDidChangeNotification, object: nil)
+    updateSaveButton()
+  }
```

Comment:
Please try to add some spacing for your codes, it will be more readable.
```suggestion
  override func viewDidLoad() {
    super.viewDidLoad()
    
    configureNavigationItem()
    setupTodoBasicInfoCell()
    
    tableView.isUserInteractionEnabled = true
    
    let notificationCenter = NotificationCenter.default
    notificationCenter.addObserver(self, selector: #selector(textViewDidChange), name: UITextView.textDidChangeNotification, object: nil)
    
    updateSaveButtonState()
  }
  
  private func configureNavigationItem() {
    navigationItem.rightBarButtonItem = saveBarButton
    
    if todo != nil {
      title = "Edit To Do"
    } else {
      title = "Add To Do"
    }
  }
  
  private func setupTodoBasicInfoCell() {
    todoBasicInfoCell.isCompleteButton.addTarget(self, action: #selector(isCompleteButtonTapped), for: .touchUpInside)
    todoBasicInfoCell.toDoTitleTextField.addTarget(self, action: #selector(toDoTitleTextFieldEditingChanged), for: .editingChanged)
  }
```